crates/sui-bridge/src/sui_client.rs

```diff
@@ -882,9 +1033,63 @@ impl SuiClientInner for sui_rpc::Client {
 
     async fn get_gas_data_panic_if_not_gas(
         &self,
-        _gas_object_id: ObjectID,
+        gas_object_id: ObjectID,
     ) -> (GasCoin, ObjectRef, Owner) {
-        todo!()
+        loop {
+            let result = async {
+                let resp = self
+                    .clone()
+                    .ledger_client()
+                    .get_object(
+                        GetObjectRequest::new(&(gas_object_id.into())).with_read_mask(
+                            FieldMask::from_paths([
+                                Object::path_builder().object_id(),
```

Comment:
Lets request the `bcs` field and then we can deser directly into the monorepo Object type.